ldap/servers/plugins/uiduniq/uid.c

```diff
@@ -820,16 +820,16 @@ searchAllSubtrees(Slapi_DN **subtrees, Slapi_DN **exclude_subtrees, const char *
 
     /*
    * For each DN in the managed list, do uniqueness checking if
-   * the target DN is a subnode in the tree.
+   * the destination (target) DN is a subnode in the tree.
    */
     for (i = 0; subtrees && subtrees[i]; i++) {
         Slapi_DN *sufdn = subtrees[i];
         /*
      * The DN should already be normalized, so we don't have to
      * worry about that here.
      */
-        if (unique_in_all_subtrees || slapi_sdn_issuffix(dn, sufdn)) {
-            result = search(sufdn, attrNames, attr, values, requiredObjectClass, dn, exclude_subtrees);
+        if (unique_in_all_subtrees || slapi_sdn_issuffix(sourceSDN, sufdn)) {
```

Comment:
No, we should check that subtree if the destinationSDN lands into that monitored sufdn